### DISCUSSION TITLE
Multithreading

### DISCUSSION THREAD

**Discussion by AndyDecker** [oakserver/oak#396] [General]
Hey,
as far as i have seen oak is singlethreaded and i wondered if multithreading is planned or if someone already is using oak in a multithreaded environment. Thanks for any feedback!

**Comment by kitsonk** [comment]
JavaScript is a single threaded language. I am not sure what you envision.

**Reply by AndyDecker** [reply]
some of my calls trigger tasks that can take some time, so having them sheduled without blocking the rest would be better for me. In a node.js environment i would normally use [Worker Threads](https://nodejs.org/api/worker_threads.html). But i'm pretty new to deno so don't know how things are handled here.

**Reply by jtoppine** [reply]
Due to Deno/Oak having highly async apis all around, they are pretty good at multitasking even without any threading/workers.  But you can use Web Workers in Oak/Deno for exactly that purpose, running tasks in parallel, if that fits your needs.

**Reply by kitsonk** [reply]
Deno supports web workers, and allows workers with the `Deno` namespace.  Offloading some intensive work to a web worker, and not responding to a request until that is done in a non-blocking way is certainly possible, but doesn't require any specific support from oak.

As mentioned by @jtoppine oak is already non-blocking in the way it handles requests it receives, so multi-threading might not be needed in a lot of use cases.
